### DISCUSSION TITLE
Default chain RPC_URL of mumbai publicProvider appears be having issues

### DISCUSSION THREAD

**Discussion by EbenKouao** [wevm/wagmi#2803] [Q&A]
The default mumbai public provider: https://matic-mumbai.chainstacklabs.com appears to be having some issues.

Current behavior: unable to interact with this RPC URL of `public` polygonMumbai
Although it's recommended to use a dedicated RPC URL [such as Alchemy / Infura](https://wagmi.sh/react/providers/public#:~:text=Only%20having%20publicProvider%20in%20your%20providers%20will%20make%20the%20chain%20use%20the%20public%20RPC%20URL%20which%20could%20lead%20to%20rate%2Dlimiting.%20It%20is%20recommended%20to%20also%20include%20another%20provider%20in%20your%20list%20(such%20as%3A%20alchemyProvider%2C%20infuraProvider%20or%20jsonRpcProvider).)

In wagmi node_modules `@wagmi/chains/dist/index.d.ts` the default public provider is `https://matic-mumbai.chainstacklabs.com`. which appears to be down (at least temporarily).

``` javascript
readonly public: {
            readonly http: readonly ["https://matic-mumbai.chainstacklabs.com"];
        };
```

Ideally, I want to confirm if this is an issue others are also facing, and the recommended best practice e.g. use an alchemy key / RPC_URL.

Cheers.

**Comment by Diego-c22** [comment]
I'm facing this, any solution?

**Comment by Diego-c22** [comment]
The solution I found is to add `Mumbai` as [custom chain](https://wagmi.sh/react/chains#build-your-own), making use of a [jsonRpcProvided](https://wagmi.sh/react/providers/jsonRpc). Here is an implementation.

**Chain**
```
import { Chain } from "wagmi";

export const mumbai = {
  id: 80_001,
  name: "Mumbai",
  network: "Mumbai",
  nativeCurrency: {
    decimals: 18,
    name: "MATIC",
    symbol: "MATIC",
  },
  rpcUrls: {
    public: { http: [mumbaiRPC},
    default: { http: [mumbaiRPC] },
  },
  blockExplorers: {
    etherscan: { name: "PolygonScan", url: "https://mumbai.polygonscan.com/" },
    default: { name: "PolygonScan", url: "https://mumbai.polygonscan.com/" },
  },
} as const satisfies Chain;
```

**Configure chain**
```
const { chains, publicClient, webSocketPublicClient } = configureChains(
  [mumbai, polygon],
  [
    alchemyProvider({ apiKey: API_KEY }),
    jsonRpcProvider({
      rpc: (chain) => ({
        http: chain.id === 80_001 ? mumbaiRPC : polygonRPC,
      }),
    }),
  ],
);
```
